internal/install/recipe_installer.go

```diff
@@ -235,7 +237,12 @@ func (i *RecipeInstaller) executeAndValidateWithProgress(ctx context.Context, m
 		fmt.Println(r.PreInstallMessage())
 	}
 
-	vars, err := i.recipeExecutor.Prepare(ctx, *m, *r, i.AssumeYes)
+	licenseKey, err := i.licenseKeyFetcher.FetchLicenseKey(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	vars, err := i.recipeExecutor.Prepare(ctx, *m, *r, i.AssumeYes, licenseKey)
```

Comment:
This `Prepare` method seems to be increasingly meaningless as a contract especially as we add more arguments to it.  Would it be better to add the licenseKeyFetcher as a dependency on the recipeExecutor instead, and do the fetching from within?